### DISCUSSION TITLE
Getting LIBUSB_TRANSFER_ERROR when working in USB 3.2 port

### DISCUSSION THREAD

**Discussion by pramod-kg-sndk** [libusb/libusb#1524] [Q&A]
I was working in Ubuntu 22.04 PC, where I have a script to send SCSI command to device in UAS mode. It is working properly in USB 2.x ports( taken care of read ready and write ready IUs), but showing LIBUSB_TRANSFER_ERROR when requesting for status IU after the command transfer complete. This is happening with Test Unit Ready command. I would like to know why it is throwing error while requesting status. 
Could some can help me to resolve this issue?

**Comment by mcuee** [comment]
Probably nothing to do with libusb but the underlying kernel.

**Reply by pramod-kg-sndk** [reply]
On further debug we could see that, device is not sending ERDY after the command phase though it is giving ACK to command on Command End Point. What could be the reason?

**Reply by pramod-kg-sndk** [reply]
Hi @mcuee ,
From the below trace it looks like host is requesting for data on stream 0, though the stream id used was 1. 
![Screenshot 2024-06-27 165013](https://github.com/libusb/libusb/assets/132047534/3fe42230-3ef1-40b0-8ccf-b4cdce5661d5)

Could this be an issue from libusb library?

**Reply by mcuee** [reply]
I still think it is not an issue with libusb. Can  you test with another PC with newer Ubuntu version or change to another PC? Thanks.

**Reply by mcuee** [reply]
BTW, why do you want to use libusb in this case? To me it is probably not the proper library for your application.

**Reply by mcuee** [reply]
@tormodvolden 

Please help to provide your comments here as well. Thanks.

**Reply by pramod-kg-sndk** [reply]
> I still think it is not an issue with libusb. Can you test with another PC with newer Ubuntu version or change to another PC? Thanks.

Tested the same in two PC with Ubuntu 22.04 and the result was same.

**Reply by pramod-kg-sndk** [reply]
> BTW, why do you want to use libusb in this case? To me it is probably not the proper library for your application.

intention is to create a framework for testing SCSI commands. Please let me know why do you think it is not suitable for this application?
